pkg/resources/metrics.go

```diff
@@ -246,6 +247,43 @@ func CreateRedisConnectivityAlert(ctx context.Context, client k8sclient.Client,
 	return pr, nil
 }
 
+
+func ReconcilePostgresFreeStorageAlerts(ctx context.Context, client k8sclient.Client, inst *v1alpha1.RHMI, cr *crov1.Postgres) error {
+	if strings.ToLower(inst.Spec.UseClusterStorage) == "true" {
+		logrus.Info("skipping postgres free storage alert creation, useClusterStorage is true")
+		return nil
+	}
+
+	alertName := "PostgresStorageWillFillIn4Hours"
+	ruleName := "postgres-storage-will-fill-in-4-hours"
+	alertDescription := "The postgres instance {{ $labels.instanceID }} for product {{  $labels.productName  }} will run of disk space in the next 4 hours"
+	labels := map[string]string{
+		"severity":    "critical",
+	}
+	alertExp := intstr.FromString("predict_linear(cro_postgres_free_storage_average[1h], 4 * 3600) <= 0")
```

Comment:
should there be something in these expressions to tie them to a product ?

Reply:
the labels tie them to the product, when they fire. This is still a WIP and we are currently working on it, so likely to change :slightly_smiling_face: